extensions/aws/processors/PutS3Object.h

```diff
@@ -0,0 +1,195 @@
+/**
+ * @file PutS3Object.h
+ * PutS3Object class declaration
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <sstream>
+#include <utility>
+#include <vector>
+#include <memory>
+#include <string>
+
+#include "aws/core/auth/AWSCredentialsProvider.h"
+
+#include "S3Wrapper.h"
+#include "core/Property.h"
+#include "core/Processor.h"
+#include "core/logging/Logger.h"
+#include "core/logging/LoggerConfiguration.h"
+#include "utils/OptionalUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace aws {
+namespace processors {
+
+namespace region {
+
+  constexpr const char *US_GOV_WEST_1 = "us-gov-west-1";
+  constexpr const char *US_EAST_1 = "us-east-1";
+  constexpr const char *US_EAST_2 = "us-east-2";
+  constexpr const char *US_WEST_1 = "us-west-1";
+  constexpr const char *US_WEST_2 = "us-west-2";
+  constexpr const char *EU_WEST_1 = "eu-west-1";
+  constexpr const char *EU_WEST_2 = "eu-west-2";
+  constexpr const char *EU_CENTRAL_1 = "eu-central-1";
+  constexpr const char *AP_SOUTH_1 = "ap-south-1";
+  constexpr const char *AP_SOUTHEAST_1 = "ap-southeast-1";
+  constexpr const char *AP_SOUTHEAST_2 = "ap-southeast-2";
+  constexpr const char *AP_NORTHEAST_1 = "ap-northeast-1";
+  constexpr const char *AP_NORTHEAST_2 = "ap-northeast-2";
+  constexpr const char *SA_EAST_1 = "sa-east-1";
+  constexpr const char *CN_NORTH_1 = "cn-north-1";
+  constexpr const char *CA_CENTRAL_1 = "ca-central-1";
+  constexpr const char *AF_SOUTH_1 = "af-south-1";
+  constexpr const char *AP_EAST_1 = "ap-east-1";
+  constexpr const char *AP_NORTHEAST_3 = "ap-northeast-3";
+  constexpr const char *CN_NORTHWEST_1 = "cn-northwest-1";
+  constexpr const char *EU_SOUTH_1 = "eu-south-1";
+  constexpr const char *EU_WEST_3 = "eu-west-3";
+  constexpr const char *EU_NORTH_1 = "eu-north-1";
+  constexpr const char *ME_SOUTH_1 = "me-south-1";
+  constexpr const char *US_GOV_EAST_1 = "us-gov-east-1";
```

Comment:
1. I think we do not indent inside namespaces (although I do not think that it hinders readability here)
2. also do you think that putting the regions in lexicographical order could help us add-remove (or diff) the list in the future?

Reply:
Fixed in [dff0c8a](https://github.com/apache/nifi-minifi-cpp/pull/915/commits/dff0c8a327c5c6e13a4f6008314986e3bd57b441)